beacon_node/network/src/sync/block_lookups/single_block_lookup.rs

```diff
@@ -343,336 +176,290 @@ impl<const MAX_ATTEMPTS: u8, T: BeaconChainTypes> SingleBlockLookup<MAX_ATTEMPTS
+    }
 
-        if self.blob_request_state.requested_ids.is_empty() {
-            return Ok(None);
-        }
+    /// Updates this request with the most recent picture of which blobs still need to be requested.
+    pub fn update_blobs_request(&mut self) {
+        self.blob_request_state.requested_ids = self.missing_blob_ids()
+    }
 
-        debug_assert!(matches!(
-            self.blob_request_state.state.state,
-            State::AwaitingDownload
-        ));
-        let request = BlobsByRootRequest {
-            blob_ids: VariableList::from(self.blob_request_state.requested_ids.clone()),
-        };
-        let response_type = ResponseType::Blob;
-        if self.too_many_attempts(response_type) {
-            Err(LookupRequestError::TooManyAttempts {
-                cannot_process: self.cannot_process(response_type),
-            })
-        } else if let Some(peer_id) = self.get_peer(response_type) {
-            self.add_used_peer(peer_id, response_type);
-            Ok(Some((peer_id.to_peer_id(), request)))
+    /// If `unknown_parent_components` is `Some`, we know block components won't hit the data
+    /// availability cache, so we don't check it. In either case we use the data availability
+    /// checker to get a picture of outstanding blob requirements for the block root.
+    pub(crate) fn missing_blob_ids(&self) -> Vec<BlobIdentifier> {
+        if let Some(components) = self.cached_child_components.as_ref() {
+            let blobs = components.downloaded_indices();
+            self.da_checker
+                .get_missing_blob_ids(
+                    self.block_root(),
+                    components.downloaded_block.as_ref(),
+                    Some(blobs),
+                )
+                .unwrap_or_default()
         } else {
-            Err(LookupRequestError::NoPeers)
+            self.da_checker
+                .get_missing_blob_ids_checking_cache(self.block_root())
+                .unwrap_or_default()
         }
     }
 
-    fn too_many_attempts(&self, response_type: ResponseType) -> bool {
-        match response_type {
-            ResponseType::Block => self.block_request_state.state.failed_attempts() >= MAX_ATTEMPTS,
-            ResponseType::Blob => self.blob_request_state.state.failed_attempts() >= MAX_ATTEMPTS,
+    /// Penalizes a blob peer if it should have blobs but didn't return them to us. Does not penalize
+    /// a peer who we request blobs from based on seeing a block or blobs over gossip. This may
+    /// have been a benign failure.
+    pub fn penalize_lazy_blob_peer(&mut self, cx: &SyncNetworkContext<T>) {
+        if let Ok(blob_peer) = self.blob_request_state.state.processing_peer() {
+            if let PeerShouldHave::BlockAndBlobs(blob_peer) = blob_peer {
+                cx.report_peer(
+                    blob_peer,
+                    PeerAction::MidToleranceError,
+                    "single_block_failure",
```

Comment:
do we differentiate between block & blob here? i.e. use `single_blob_failure`

Reply:
ah ignore this - looks like they're now requested together and it's just a single block lookup request